utils/text_extractor.py

```diff
@@ -52,6 +56,8 @@ def get_all_text_from_url(url, retries=5):
             logging.error("an exception occurred while trying to access url {} trying again \n more details: {}".format(url, exc))
             time.sleep(1)
             retries = retries - 1
+            if retries == 0 and type(exc).__name__ == "timeout":
+                increase_timeout_count(url)
```

Comment:
you can use Python "expect" specific type instead of dealing with type extraction.
by doing so you will make the code more clean and readable